packages/paste-core/components/form/src/FormLabel.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+import css from '@styled-system/css';
+import {Box} from '@twilio-paste/box';
+import {Flex} from '@twilio-paste/flex';
+import {Text, safelySpreadTextProps} from '@twilio-paste/text';
+
+export interface FormLabelProps extends React.LabelHTMLAttributes<''> {
+  htmlFor: string;
+  disabled?: boolean;
+  required?: boolean;
+  marginBottom?: 'space0';
+  className?: never;
+  style?: never;
+}
+
+// eslint-disable-next-line emotion/syntax-preference
+const StyledRequiredDot = styled(Box)(() =>
+  css({
+    display: 'block',
+    width: '4px',
+    height: '4px',
+    marginRight: 'space20',
+    backgroundColor: 'colorBackgroundRequired',
+    borderRadius: '50%',
+  })
+);
+
+const FormLabel: React.FC<FormLabelProps> = ({marginBottom, required, disabled, children, ...props}) => {
+  return (
+    <Text
+      {...safelySpreadTextProps(props)}
+      as="label"
+      display="inline-block"
```

Comment:
This should probably be `block` so that it spans the full width. That was I get the full row as a click target, no just the text

Reply:
It'll also get rid of this inline-block problem when adding the dot, which cause a few pixel jump in size of the parent.

![](https://p13.f2.n0.cdn.getcloudapp.com/items/OAurP5Lk/Screen+Recording+2020-03-06+at+17.09.gif?v=48d14e23cf4277b41fef13f77a77482c)

Reply:
It could be `display: flex`, but `alignItems` isn't supported on `Text`.

Reply:
I worked out a different way to use `Flex` and `Text`.